### PR TITLE
Enable boost python

### PR DESCRIPTION
**Pull request by asa**
Bring in boost python. 
This works for my config. perhaps its a useful base for others.

### CODE REVIEW

File: boost/boost.bzl [hunk 1]
```diff
@@ -15,7 +15,7 @@ default_defines = select({
 def srcs_list(library_name, exclude):
     return native.glob(
         ["libs/%s/src/*" % library_name],
-        exclude = ["**/*.asm", "**/*.S", "**/*.doc"] + exclude,
+        exclude = ["**/*.asm", "**/*.S", "**/*.doc","**/fabscript"] + exclude,
```
Comment (nelhage) [inline boost/boost.bzl:18]: Should we pass this just for the `python` library instead of globally?

**Comment by asa** [comment]
Yep, agreed. Pushed.

**Comment by wolfd** [comment]
This can be subtly wrong depending on how you compile. It would be much safer to use `boost_so_library` here:
https://github.com/nelhage/rules_boost/issues/67#issuecomment-2015272810

File: boost.BUILD [hunk 2]
```diff
@@ -996,6 +996,23 @@ boost_library(
     ],
 )
 
+boost_library(
+    name = "python",
+    exclude_src = ["**/fabscript"],
```
Comment (wolfd) [inline boost.BUILD:1001]: Why not `exclude_src = ["libs/python/src/fabscript"],`?

Also, nitpick, but there's trailing whitespace on this line

**Pull request merged**